lighthouse-core/test/lib/network-request-test.js

```diff
@@ -6,189 +6,170 @@
 'use strict';
 
 const NetworkRequest = require('../../lib/network-request.js');
-const assert = require('assert');
+const NetworkRecorder = require('../../lib/network-recorder.js');
+const networkRecordsToDevtoolsLog = require('../network-records-to-devtools-log.js');
 
 /* eslint-env jest */
-describe('network request', function() {
+describe('NetworkRequest', () => {
   afterEach(() => {
     global.isLightrider = undefined;
   });
 
-  function getRequest() {
-    const req = new NetworkRequest();
-    req.transferSize = 100;
-    req.responseHeaders = [{name: NetworkRequest.HEADER_FETCHED_SIZE, value: 10}];
-    return req;
-  }
+  describe('update transfer size for Lightrider', () => {
+    function getRequest() {
+      return {
+        transferSize: 100,
+        responseHeaders: [{name: NetworkRequest.HEADER_FETCHED_SIZE, value: '10'}],
+      };
+    }
 
-  describe('update transfer size for lightrider', function() {
-    it('does nothing if not Lightrider', function() {
+    it('does nothing if not Lightrider', () => {
       const req = getRequest();
-      global.isLightrider = false;
+      expect(req.transferSize).toBe(100);
 
-      assert.equal(req.transferSize, 100);
-      req._updateTransferSizeForLightrider();
-      assert.equal(req.transferSize, 100);
+      const devtoolsLog = networkRecordsToDevtoolsLog([req]);
+      const record = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
+
+      expect(record.transferSize).toBe(100);
     });
 
-    it('updates transfer size', function() {
+    it('updates transfer size if Lightrider', () => {
       const req = getRequest();
+      expect(req.transferSize).toBe(100);
+
+      const devtoolsLog = networkRecordsToDevtoolsLog([req]);
       global.isLightrider = true;
+      const record = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
 
-      assert.equal(req.transferSize, 100);
-      req._updateTransferSizeForLightrider();
-      assert.equal(req.transferSize, 10);
+      expect(record.transferSize).toBe(10);
     });
 
-    it('does nothing if header is non float', function() {
+    it('does nothing if header is non float', () => {
       const req = getRequest();
+      req.responseHeaders = [
+        {name: NetworkRequest.HEADER_FETCHED_SIZE, value: 'ten'},
+      ];
+      expect(req.transferSize).toBe(100);
+
+      const devtoolsLog = networkRecordsToDevtoolsLog([req]);
       global.isLightrider = true;
-      req.responseHeaders = [{name: NetworkRequest.HEADER_FETCHED_SIZE, value: 'ten'}];
+      const record = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
 
-      assert.equal(req.transferSize, 100);
-      req._updateTransferSizeForLightrider();
-      assert.equal(req.transferSize, 100);
-      assert.notEqual(req.transferSize, NaN);
+      expect(record).toMatchObject(req);
```

Comment:
nit: I think it'd be clearer if it was just like the others `expect(req.transferSize).toBe(100)`